android/src/com/mapswithme/maps/MwmApplication.java

```diff
@@ -161,7 +161,43 @@ public String getApkPath()
     }
   }
 
-  public String getDataStoragePath()
+  private static boolean isMapStorage(String path)
+  {
+    final File dir = new File(path);
+    if (TextUtils.isEmpty(path) || !Utils.isDirectoryReadable(dir))
+      return false;
+
+    // Current map directory should contain folders with map indices at least, so use this fact as an indicator.
+    final File[] list = dir.listFiles(new FileFilter()
+    {
+      @Override
+      public boolean accept(File file)
+      {
+        return file.isDirectory();
+      }
+    });
+
+    return (list.length != 0);
+  }
+
+  private String findMapStoragePath()
+  {
+    String res = mPrefs.getString(Constants.STORAGE_PATH_PREF, "");
```

Comment:
We have already discussed, that Android preferences are not needed - native are well enough.